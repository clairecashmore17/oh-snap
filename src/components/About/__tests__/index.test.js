import React from "react";

//retrieve function from the React Testing LIbrary
import { render, cleanup } from "@testing-library/react";

// extend-expect package, offers access to custom matchers that are used to test DOM elements
import "@testing-library/jest-dom/extend-expect";

import About from "..";
const { asFragment } = render(<About />);
//Configure our test environment

//then use describe function to decalre the component we are testing
describe("About component", () => {
  //First Test
  // it function : test and it are interchangeable
  it("render", () => {
    render(<About />);
  });
  //Second Test
  // snapshot is a serialized version of the DOM node structure enabled by jest
  it("matches snapshot DOM note structure", () => {
    //Comparing whether the expected and actual outcomes match
    expect(asFragment()).toMatchSnapshot();
  });
});

// ensures that after each test, no leftover memory data that could give false results
afterEach(cleanup);
