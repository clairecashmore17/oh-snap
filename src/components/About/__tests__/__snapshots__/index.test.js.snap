// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About component matches snapshot DOM note structure 1`] = `<DocumentFragment />`;
